cmd/node/main.go

```diff
@@ -547,17 +560,24 @@ func startNode(ctx *cli.Context, log *logger.Logger, version string) error {
 		return err
 	}
 
-	ef := facade.NewElrondNodeFacade(currentNode, apiResolver)
+	if currentNode.GetAppStatusHandler() != nil {
+		pollingInterval := time.Duration(generalConfig.GeneralSettings.StatusPollingIntervalSec) * time.Second
+		appStatusPollingHandler, err := appStatusPolling.NewAppStatusPolling(currentNode.GetAppStatusHandler(),
+			pollingInterval)
 
-	prometheusURLAvailable := true
-	prometheusJoinUrl, err := getPrometheusJoinURL(ctx.GlobalString(serversConfigurationFile.Name))
-	if err != nil || prometheusJoinUrl == "" {
-		prometheusURLAvailable = false
+		if err == nil {
```

Comment:
We should at least log the errors here. And might need a small refactor (extract in another method) since we have 3 levels of indentation here. Having this in another function would allow to early exit and have cleaner code:

```go
appStatusPollingHandler, err := appStatusPolling.NewAppStatusPolling(currentNode.GetAppStatusHandler(),
			pollingInterval)

if err != nil {
    // Log error
    return
}
err = appStatusPollingHandler.SetConnectedAddresses(networkComponents.NetMessenger)
// Again :)
```

Reply:
logged errors and moved code to a function